lib/identity/client.js

```diff
@@ -694,13 +686,66 @@ class Client extends PartialClient {
         currentGroup.groupID,
         recordType
       )
-      if (secretType === 'File') {
-        await this.storage.revoke(recordType, clientID)
-      }
       return revokedFromGroup
     }
   }
 
+  /**
+   * removeIdentityFromNamespace takes a user defined namespace name and a username for an identity and removes access to the given namespace for the identity
+   *
+   * @param {String} namespace The user given name for the secret stored in the meta data
+   * @param {String} userToRevokeShare The username of the user that will have permissions revoked
+   *
+   * @return {Boolean}
+   */
+  async removeIdentityFromNamespace(userToRevokeShare, namespace) {
+    // Validation
+    if (userToRevokeShare == '') {
+      throw new Error('Username Required')
+    }
+    // Look Up the ClientID for Username
+    let clientID = await this.clientIDLookUpByUsername(userToRevokeShare)
+    if (clientID === null) {
+      return null
+    }
+    // Look up group for our current namespace
+    let currentGroup = await this.getNamespace(namespace)
+    // If namespace doesn't exist return true
+    if (currentGroup.groupID === undefined) {
+      return true
+    }
+    // Remove identity from namespace
+    let revokeFromNamespace = await this.storage.removeGroupMembers(
```

Comment:
In JS, instead of awaiting into a variable, and then immediately returning that variable, just return the function call. It doesn't matter much, but the difference can add up in wrapper land.

return as written:

- extra variable memory location to point to the promise from removeGroupMembers
- awaiting at line 718 makes the return Promise<Promise<removeGroupMemebers()>>

return directly
- async determines if the return is plain or a promis, and wraps the plain return in a promise as needed
- `return this.storage.removeGroupMembers(...)`
- Return is Promise<removeGroupMemebers()>